src/llms/azure.ts

```diff
@@ -0,0 +1,345 @@
+  return pipe(
+    extractContent(),
+    scan((acc, curr) => acc + curr, ''),
+  );
+}
+
+/**
+ * Make a request to OpenAI's chat-completions API via the Grafana LLM plugin proxy.
+ */
+export async function chatCompletions(request: ChatCompletionsRequest): Promise<ChatCompletionsResponse> {
+  const response = await getBackendSrv().post<ChatCompletionsResponse>(`${LLM_PLUGIN_ID}/resources/${AZURE_OPENAI_CHAT_COMPLETIONS_PATH}`, request, {
+    headers: { 'Content-Type': 'application/json' }
+  });
+  return response;
+}
+
+/**
+ * Make a streaming request to OpenAI's chat-completions API via the Grafana LLM plugin proxy.
+ *
+ * A stream of tokens will be returned as an `Observable<string>`. Use the `extractContent` operator to
+ * filter the stream to only content messages, or the `accumulateContent` operator to obtain a stream of
+ * accumulated content messages.
+ *
+ * The 'done' message will not be emitted; the stream will simply end when this message is encountered.
+ *
+ * @example <caption>Example of reading all tokens in a stream.</caption>
+ * const stream = streamChatCompletions({ model: 'gpt-3.5-turbo', messages: [
+ *   { role: 'system', content: 'You are a great bot.' },
+ *   { role: 'user', content: 'Hello, bot.' },
+ * ]}).pipe(extractContent());
+ * stream.subscribe(console.log);
+ * // Output:
+ * // ['Hello', '? ', 'How ', 'are ', 'you', '?']
+ *
+ * @example <caption>Example of accumulating tokens in a stream.</caption>
+ * const stream = streamChatCompletions({ model: 'gpt-3.5-turbo', messages: [
+ *   { role: 'system', content: 'You are a great bot.' },
+ *   { role: 'user', content: 'Hello, bot.' },
+ * ]}).pipe(accumulateContent());
+ * stream.subscribe(console.log);
+ * // Output:
+ * // ['Hello', 'Hello! ', 'Hello! How ', 'Hello! How are ', 'Hello! How are you', 'Hello! How are you?']
+ */
+export function streamChatCompletions(request: ChatCompletionsRequest): Observable<ChatCompletionsResponse<ChatCompletionsChunk>> {
+  const channel: LiveChannelAddress = {
+    scope: LiveChannelScope.Plugin,
+    namespace: LLM_PLUGIN_ID,
+    path: AZURE_OPENAI_CHAT_COMPLETIONS_PATH,
+    data: request,
+  };
+  const messages = getGrafanaLiveSrv()
+    .getStream(channel)
+    .pipe(filter((event) => isLiveChannelMessageEvent(event))) as Observable<LiveChannelMessageEvent<ChatCompletionsResponse<ChatCompletionsChunk>>>
+  return messages.pipe(
+    takeWhile((event) => !isDoneMessage(event.message.choices[0].delta)),
+    map((event) => event.message),
+  );
+}
+
+let loggedWarning = false;
+
+/** Check if the OpenAI API is enabled via the LLM plugin. */
+export const enabled = async () => {
+  try {
+    const settings = await getBackendSrv().get(`${LLM_PLUGIN_ROUTE}/settings`, undefined, undefined, {
+      showSuccessAlert: false, showErrorAlert: false,
+    });
+    return settings.enabled && (settings?.secureJsonFields?.openAIKey ?? false);
```

Comment:
We'll need to check a different field here since we're using Azure.

```suggestion
    return settings.enabled && (settings?.secureJsonFields?.azureOpenAIKey ?? false);
```

Having said that, I'd quite like to refactor this `enabled` logic into the LLM plugin as a healthcheck endpoint... that'll  be a separate PR though.